src/java.desktop/share/native/common/awt/utility/sizecalc.h

```diff
@@ -116,7 +112,7 @@
  *    // Use the allocated memory...
  */
 #define SAFE_SIZE_STRUCT_ALLOC(func, a, m, n) \
-    (IS_SAFE_STRUCT_SIZE((a), (m), (n)) ? ((func)((a) + (m) * (n))) : FAILURE_RESULT)
+    (IS_SAFE_STRUCT_SIZE((a), (m), (n)) ? ((func)((a) + (size_t)(m) * (size_t)(n))) : FAILURE_RESULT)
```

Comment:
```suggestion
    (IS_SAFE_STRUCT_SIZE((a), (m), (n)) ? ((func)((size_t)(a) + (size_t)(m) * (size_t)(n))) : FAILURE_RESULT)
```
To be safe, `a` should also be cast.

And `IS_SAFE_STRUCT_SIZE` should also be updated to pass `(size_t)(m) * (size_t)(n)` to `IS_SAFE_SIZE_ADD` instead of `(m) * (n)`.

Reply:
> To be safe, `a` should also be cast.
>
>And IS_SAFE_STRUCT_SIZE should also be updated to pass (size_t)(m) * (size_t)(n) to IS_SAFE_SIZE_ADD instead of (m) * (n).

Sounds reasonable.